miden-crypto/src/dsa/eddsa_25519/mod.rs

```diff
@@ -132,6 +132,149 @@ impl PublicKey {
+    ///    binding.
+    ///
+    /// 3. **Bypasses Standard Protocol**: If the hash is not computed correctly as `SHA-512(R || A
+    ///    || message)`, this method bypasses standard Ed25519 verification and the signature will
+    ///    not be compatible with Ed25519 semantics.
+    ///
+    /// ## When This Might Be Used:
+    ///
+    /// This method is only appropriate in very specific scenarios where:
+    /// - You have a trusted computation environment that computes the hash correctly as `SHA-512(R
+    ///   || A || message)` (see `compute_challenge_k()`)
+    /// - You need to separate the hashing phase from the EC verification phase (e.g., for different
+    ///   execution environments or performance optimization)
+    /// - You fully understand the security implications and have a threat model that accounts for
+    ///   them
+    ///
+    /// When the hash is computed correctly, this method implements standard Ed25519 verification.
+    ///
+    /// ## Standard Usage:
+    ///
+    /// For normal Ed25519 verification, use `verify()` instead.
+    ///
+    /// # Arguments
+    /// * `k_hash` - A 64-byte hash (typically computed as `SHA-512(R || A || message)`)
+    /// * `signature` - The signature to verify
+    ///
+    /// # Returns
+    /// `true` if the verification equation `[s]B = R + [k]A` holds, `false` otherwise
+    ///
+    /// # Warning
+    /// Do NOT use this method unless you fully understand Ed25519's cryptographic properties
+    /// and have a specific need for this low-level operation.
+    pub fn verify_with_k(&self, k_hash: [u8; 64], signature: &Signature) -> bool {
+        use curve25519_dalek::{
+            edwards::{CompressedEdwardsY, EdwardsPoint},
+            scalar::Scalar,
+            traits::IsIdentity,
+        };
+
+        // Reduce the 64-byte hash modulo L to get the challenge scalar
+        let k_scalar = Scalar::from_bytes_mod_order_wide(&k_hash);
+
+        // Extract signature components: R (first 32 bytes) and s (second 32 bytes)
+        let sig_bytes = signature.inner.to_bytes();
+        let r_compressed = CompressedEdwardsY(
+            sig_bytes[..32].try_into().expect("signature R component is exactly 32 bytes"),
+        );
+        let s_bytes: [u8; 32] =
+            sig_bytes[32..].try_into().expect("signature s component is exactly 32 bytes");
+
+        // Decompress R point
+        let r_point = match r_compressed.decompress() {
+            Some(point) => point,
+            None => return false, // Invalid R point
+        };
+
+        // Convert s bytes to Scalar
+        let s = Scalar::from_bytes_mod_order(s_bytes);
+
+        // Get public key as EdwardsPoint
+        let a_compressed = CompressedEdwardsY(self.inner.to_bytes());
+        let a_point = match a_compressed.decompress() {
+            Some(point) => point,
+            None => return false, // Invalid public key
```

Comment:
Properly speaking I believe this should be an `unreachable!` given the `VerifyingKey` guarantees on `self.inner`.